front/src/pods/trainer/trainer.container.tsx

```diff
@@ -20,28 +21,26 @@ interface Params {
 export const TrainerContainer = () => {
   const { token, room } = useParams<Params>();
   const { log, appendToLog, logRef } = useLog();
-  const [socket, setSocket, socketRef] = useWithRef<SocketIO.Socket>(null);
+  const socketDetails = { room: room, trainertoken: token };
+  const [socket, setSocket, socketRef] = useWithRef<SocketIO.Socket>(createSocket(socketDetails));
 
   const [currentTrainerUrl, setCurrentTrainerUrl] = React.useState<string>('');
   const [currentStudentUrl, setCurrentStudentUrl] = React.useState<string>('');
 
   const handleConnection = () => {
-    // Connect to socket
-    const localSocket = createSocket({
-      room: room,
-      trainertoken: token,
-    });
-
-    setSocket(localSocket);
-
-    localSocket.on(SocketOuputMessageLiteral.MESSAGE, msg => {
+    socket.on(SocketOuputMessageLiteral.MESSAGE, msg => {
       if (msg.type) {
         const { type, payload } = msg;
 
         switch (type) {
           case SocketReceiveMessageTypes.APPEND_TEXT:
             appendToLog(payload);
             break;
+          case SocketReceiveMessageTypes.TRAINER_GET_FULL_CONTENT:
+            appendToLog(payload);
```

Comment:
I think it would be a good idea to replace the log in this case instead of append it (altough in the initial case we are using the log is empty).

Reply:
I agree, it has been fixed in the pull request number #71 that solves the issue #4.

Reply:
please introduce the fix in this branch, each case should be isolated, give me a buzz when it's ready on this branch

Reply:
The branch #53 has been updated with that fix. Please, check it out.